acmeclient.go

```diff
@@ -94,10 +94,6 @@ func (am *ACMEManager) newACMEClientWithAccount(ctx context.Context, useTestCA,
 					}
 				}
 			}
-		} else {
-			// can't prompt a user who isn't there; they should
-			// have reviewed the terms beforehand
-			am.Agreed = true
```

Comment:
In the past we've been bitten in major ways by being less careful with this logic; I think we should leave this unchanged unless there's a specific case that is breaking with this here.

Reply:
About not doing the proposed change, [here](https://github.com/caddyserver/certmagic/pull/165/commits/c69105db56649b3bc3f86007d5b8bb0e2dcdcd9c#diff-2ab2d41f53c606ee1562d2abbf29ccc2058fda9d1d510f5dc68b04bf06f6ec6eR233-R260) is the workaround I had to do without it. It is messy to say the least.

However I think it is better if this actually breaks, because it only happens if the user did not accept the TOS, i.e. the current method assumes that the TOS is accepted regardless (hence the need for this PR). Yes, we do need to check the other testers and add `CheckAccountTOS` call beforehand to make sure they are automated properly. If you can point me to some of the potential places where it needs I can add appropriate test. Also can we add something like `step_ca` to the go tests to automate all the necessary tests?